src/include/packing.i

```diff
@@ -684,6 +686,8 @@ __wt_struct_unpackv(WT_SESSION_IMPL *session,
 
 	WT_RET(__pack_init(session, &pack, fmt));
 	while ((ret = __pack_next(&pack, &pv)) == 0) {
+		if (p >= end && pv.type != 'u')
+			return (ENOMEM);
 		WT_RET(__unpack_read(session, &pv, &p, (size_t)(end - p)));
```

Comment:
@michaelcahill, in this case, __unpack_read does not detect that there aren't enough bytes when `end == p`. In that case, a zero size is passed to `__unpack_read`, and `WT_SIZE_CHECK_UNPACK` is called with second arg being 0.  That is allowed, presumably a zero size means no checks.

I'm up against my vacation (leaving tomorrow AM), probably won't get to work on this until I return. Nobody is waiting on this pull request AFAIK.